app/src/main/java/com/duckduckgo/app/reinstalls/DownloadsDirectoryManager.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.reinstalls
+
+import android.os.Environment
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import java.io.File
+import javax.inject.Inject
+import timber.log.Timber
+
+interface DownloadsDirectoryManager {
+
+    fun getDownloadsDirectory(): File
+    fun createNewDirectory(directoryName: String)
+}
+
+@ContributesBinding(AppScope::class)
+class DownloadsDirectoryManagerImpl @Inject constructor() : DownloadsDirectoryManager {
+
+    override fun getDownloadsDirectory(): File {
```

Comment:
I'd probably move these two methods inside the `ReinstallAtbListener` and make them private there

Reply:
I put it in another class so I can mock the files in the Downloads folder and test all use cases. Does it make sense? Maybe there is a better way that I missed.

Reply:
Yeah that's fine, no strong opinion